tests/end-to-end/upgrade/pkg/tests/eventmesh-migration/helpers/application.go

```diff
@@ -0,0 +1,145 @@
+package helpers
+
+import (
+	"fmt"
+
+	"github.com/avast/retry-go"
+
+	scv1beta1 "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	servicecatalog "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+
+	appbrokerv1alpha1 "github.com/kyma-project/kyma/components/application-broker/pkg/apis/applicationconnector/v1alpha1"
+	appbroker "github.com/kyma-project/kyma/components/application-broker/pkg/client/clientset/versioned"
+	appconnectorv1alpha1 "github.com/kyma-project/kyma/components/application-operator/pkg/apis/applicationconnector/v1alpha1"
+	appconnector "github.com/kyma-project/kyma/components/application-operator/pkg/client/clientset/versioned"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	messagingclientv1alpha1 "knative.dev/eventing/pkg/client/clientset/versioned/typed/messaging/v1alpha1"
+	serving "knative.dev/serving/pkg/client/clientset/versioned"
+)
+
+const (
+	integrationNamespace = "kyma-integration"
+)
+
+type ApplicationOption func(*appconnectorv1alpha1.Application)
+
+func CreateApplication(appConnectorInterface appconnector.Interface, name string, applicationOptions ...ApplicationOption) error {
+	application := &appconnectorv1alpha1.Application{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
Why do you define a TypeMeta in a Go object? TypeMeta is only necessary when serialization to/from JSON is involved.

Reply:
removed